typeguard/__init__.py

```diff
@@ -813,11 +813,12 @@ def wrapper(*args, **kwargs):
         # If a generator is returned, wrap it if its yield/send/return types can be checked
         if inspect.isgenerator(retval) or isasyncgen(retval):
             return_type = memo.type_hints.get('return')
-            origin = getattr(return_type, '__origin__')
-            if origin in generator_origin_types:
-                return TypeCheckedGenerator(retval, memo)
-            elif origin is not None and origin in asyncgen_origin_types:
-                return TypeCheckedAsyncGenerator(retval, memo)
+            if return_type:
```

Comment:
What if `return_type` is `None`?

Reply:
I think it would have to be "NoneType" rather than "None".

Reply:
In Python 3, `NoneType` is not importable. PEP 484 states that `None` is used in its place, so functions returning `None` will have `None` as the return type.

Reply:
```python
import typing


def foo() -> None:
    pass


print(typing.get_type_hints(foo))  # prints {'return': <class 'NoneType'>}
```

So @prescod is right that the check should work correctly.